lib/ansible/modules/packaging/os/swupd.py

```diff
@@ -0,0 +1,324 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: swupd
+short_description: Manages bundles with M(swupd).
+description:
+  - Manages bundles with the M(swupd) update manager, which is used by the
+    Clear Linux Project for Intel Architecture.
+version_added: "2.3"
+author: Alberto Murillo (@albertomurillo)
+options:
+  contenturl:
+    description:
+      - URL for content file download.
+    required: false
+    default: null
+  format:
+    description:
+      - The format suffix for version file downloads.
+    required: false
+    default: null
+  manifest:
+    description:
+      - Manifest to verify against to.
+    required: false
+    default: null
+    aliases: [release, version]
+  name:
+    description:
+      - Name of the (I)bundle to install or remove.
+    required: false
+    default: null
+    aliases: [bundle]
+  state:
+    description:
+      - Indicates the desired (I)bundle state. C(present) ensures the bundle
+        is installed while C(absent) ensures the (I)bundle is not installed.
+    required: false
+    default: present
+    choices: [present, absent]
+  update:
+    description:
+      - Updates the OS to the latest version.
+    required: false
+    default: no
+  url:
+    description:
+      - Overrides both I(contenturl) and I(versionurl).
+    required: false
+    default: null
+  verify:
+    description:
+      - Verify content for OS version.
+    required: false
+    default: null
+  versionurl:
+    description:
+      - URL for version string download.
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+- name Update the OS to the latest version
+  swupd:
+    update: yes
+
+- name: Installs the "foo" bundle
+  swupd:
+    name: foo
+    state: present
+
+- name: Removes the "foo" bundle
+  swupd:
+    name: foo
+    state: absent
+
+- name: Check integrity of filesystem
+  swupd:
+    verify: yes
+
+- name: Downgrade OS to release 12920
+  swupd:
+    verify: yes
+    manifest: 12920
+'''
+
+
+class Swupd():
+    FILES_NOT_MATCH = "files did not match"
+    FILES_REPLACED = "missing files were replaced"
+    FILES_FIXED = "files were fixed"
+    FILES_DELETED = "files were deleted"
+
+    def __init__(self, module):
+        # Fail if swupd is not found
+        self.swupd_cmd = module.get_bin_path("swupd", False)
+        if not self.swupd_cmd:
+            module.fail_json(msg="Could not find swupd.")
+
+        # Initialize parameters
+        self.contenturl = module.params["contenturl"]
```

Comment:
Just looking at this, it could be simplified by using something like this:

```
# Create shortcuts
for key in module.params.keys():
  setattr(self, key, module.params[key])
```